lib/platforms/platform.ts

```diff
@@ -0,0 +1,82 @@
+import { Message, MessageTypes } from './message';
+import { Logger } from './../logger';
+import diff from '../diff';
+import * as q from 'q';
+let logger = new Logger('Platform');
+export enum GroupTypes {
+    None,
+    PUBLIC,
+    PRIVATE,
+    PERSONAL
+}
+
+export class Platform {
+    protected normalizers = [];
+    protected messageTypes = {
+        FILE_SHARE: 'file_share'
+    };
+    public defaultRecipient: string;
+    public defaultMention: string;
+    constructor(...args) {
+        this.createNormalizers();
+    }
+
+    auth(): PromiseLike<Platform> {
+        return q.resolve(this);
+    }
+    normalize(inputObj, additionalNormalizer?: (msg: any) => any) {
+        this.normalizers.forEach(normalizer => {
+            inputObj = normalizer(inputObj);
+        });
+        if (additionalNormalizer) {
+            return additionalNormalizer(inputObj);
+        }
+        return inputObj;
+    }
+
+    normalizeTarget(target: string | any): Object {
+        if (typeof target === 'string' || typeof target === 'number') {
+            target = {
+                text: target.toString()
+            };    
+        }
+        target.text = target.text || '';
+        target.attachments = target.attachments || [];
+        return target;
+    }
+
+    compare(
+        src: any,
+        target: string|object,
+        additionalNormalizer?: (msg: any) => any,
+        skipNormalize?: boolean) : any {
+        let normalizedSrc = this.normalize(src, additionalNormalizer);
```

Comment:
Looks like this is unused, should we add a condition on this boolean to perform normalization?